pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -382,7 +382,7 @@ func (s *DiscoveryServer) ConfigDump(w http.ResponseWriter, req *http.Request) {
 		adsClientsMutex.RLock()
 		defer adsClientsMutex.RUnlock()
 		connections, ok := adsSidecarIDConnectionsMap[proxyID]
-		if !ok {
+		if !ok || len(connections) == 0 {
```

Comment:
Not panic here, so this does not help I think.

```
istio-pilot-7487d6958d-rtvhh discovery istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).configDump(0xc420611080, 0x0, 0x0, 0x0, 0x29e48c0)
istio-pilot-7487d6958d-rtvhh discovery 	/workspace/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/ads.go:263 +0x68
```

Reply:
Check the logic.. it's panic down the road when connections map is empty

Reply:
IC, maybe a a mem leak. It's better if we can fix it.

Reply:
its better to have this check just in case doesn't it?